src/util/test_util.rs

```diff
@@ -0,0 +1,13 @@
+//! Utility functions that are useful for tests.
+
+#[inline(always)]
+/// Compare two `f32`s, and assert equality.
+pub fn assert_f32_equal(a: f32, b: f32) {
+    assert!((a - b).abs() < std::f32::EPSILON);
+}
```

Comment:
Many float comparisons have been changed to compare explicitly with EPSILON. If we have a function for it, we should use that in all relevant places. Also, rename the function to make it clear that the comparison is approximate.